File: src/Calculator/Converters/RadixToStringConverter.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include "RadixToStringConverter.h"
#include "CalcManager/Header Files/RadixType.h"
#include "CalcViewModel/Common/AppResourceProvider.h"

using namespace Platform;
using namespace Windows::Foundation;
using namespace Windows::UI::Xaml::Interop;
using namespace Windows::ApplicationModel::Resources;

namespace CalculatorApp
{
    namespace Converters
    {
        Object ^ RadixToStringConverter::Convert(Object ^ value, TypeName targetType, Object ^ parameter, String ^ language)
        {
            (void)targetType; // Unused parameter
            (void)parameter;  // Unused parameter
            (void)language;   // Unused parameter

            auto boxedInt = dynamic_cast<Box<int> ^>(value);
            Platform::String ^ convertedValue;
            auto resourceLoader = AppResourceProvider::GetInstance();
            switch (boxedInt->Value)
            {
            case RadixType::Binary:
            {
                convertedValue = resourceLoader->GetResourceString("Bin");
                break;
            }
            case RadixType::Octal:
            {
                convertedValue = resourceLoader->GetResourceString("Oct");
                break;
            }
            case RadixType::Decimal:
            {
                convertedValue = resourceLoader->GetResourceString("Dec");
                break;
            }
            case RadixType::Hex:
            {
                convertedValue = resourceLoader->GetResourceString("Hex");
                break;
            }
            default:
                break;
            };
            return convertedValue;
        }

        Object ^ RadixToStringConverter::ConvertBack(Object ^ value, TypeName targetType, Object ^ parameter, String ^ language)
        {
            throw ref new NotImplementedException();
        }

    }
}

File: src/Calculator/Converters/RadixToStringConverter.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using CalculatorApp.ViewModel.Common;

using System;

namespace CalculatorApp
{
    namespace Converters
    {
        /// <summary>
        /// Value converter that translates true to false and vice versa.
        /// </summary>
        [Windows.Foundation.Metadata.WebHostHidden]
        public sealed class RadixToStringConverter : Windows.UI.Xaml.Data.IValueConverter
        {
            public object Convert(object value, Type targetType, object parameter, string language)
            {
                var boxedInt = (value as int?);
                string convertedValue = null;
                var resourceLoader = AppResourceProvider.GetInstance();
                switch (boxedInt.Value)
                {
                    case (int)RadixType.Binary:
                        {
                            convertedValue = resourceLoader.GetResourceString("Bin");
                            break;
                        }
                    case (int)RadixType.Octal:
                        {
                            convertedValue = resourceLoader.GetResourceString("Oct");
                            break;
                        }
                    case (int)RadixType.Decimal:
                        {
                            convertedValue = resourceLoader.GetResourceString("Dec");
                            break;
                        }
                    case (int)RadixType.Hex:
                        {
                            convertedValue = resourceLoader.GetResourceString("Hex");
                            break;
                        }
                    default:
                        break;
                }

                return convertedValue;
            }

            public object ConvertBack(object value, Type targetType, object parameter, string language)
            {
                throw new NotImplementedException();
            }
        }
    }
}

